src/python/pants/jvm/util_rules_test.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import hashlib
+
+import pytest
+
+from pants.engine.fs import EMPTY_FILE_DIGEST, CreateDigest, Digest, FileContent, FileDigest
+from pants.jvm.util_rules import ExtractFileDigest
+from pants.jvm.util_rules import rules as util_rules
+from pants.testutil.rule_runner import QueryRule, RuleRunner
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        preserve_tmpdirs=True,
+        rules=[
+            *util_rules(),
+            QueryRule(FileDigest, (ExtractFileDigest,)),
+        ],
+    )
+
+
+def get_digest(rule_runner: RuleRunner, source_files: dict[str, str]) -> Digest:
+    files = [FileContent(path, content.encode()) for path, content in source_files.items()]
+    return rule_runner.request(Digest, [CreateDigest(files)])
```

Comment:
Reminder to myself: we should factor this up into `RuleRunner`. We've now copied and pasted this code >8 times.